docs/content/architecture.html.md

```diff
@@ -0,0 +1,198 @@
+The Flynn architecture is designed to be simple and understandable. Most of the
+components of Flynn are no different than the applications that are deployed on
+top of Flynn.
+
+Flynn's primary goal is to provide a solid platform to deploy applications on
+that is highly available, easy to run and manage, and requires minimal
+configuration.
+
+## Building Blocks
+
+Flynn is almost entirely written in Go. Go is one of the few languages designed
+with concurrency as a first-class feature which makes it a great fit for Flynn's
+distributed systems. Go is statically typed and reasonably fast to compile so we
+catch more bugs before they ship. The limited syntax and built-in formatting
+tools make contribution and collaboration smoother than any other language we've
+used.
+
+JSON over HTTP is used for communication, both internal and external. Our
+experience is that it is easy to debug and understand HTTP-based services. We
+use [Server-Sent Events](https://www.w3.org/TR/eventsource/) to provide streams
+over HTTP. Since most developers are already familiar with HTTP and almost every
+language already has great HTTP support, taking advantage of Flynn's APIs is
+easier than yet another RPC protocol.
+
+We target the latest LTS release of Ubuntu amd64 as our base operating system.
```

Comment:
The latest LTS version is changing in the next few days, should we be more specific about which release we are using, or just switch to the latest release soon after it becomes available?